src/App.tsx

```diff
@@ -1,18 +1,130 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useState,
+} from 'react';
+import {
+  getTodos,
+  addTodo,
+  CustomTodo,
+  deleteTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import Form from './components/Form/Form';
+import TodoList from './components/TodoList/TodoList';
+import Warning from './components/Warning/warning';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todosFromTheServer, setTodosFromTheServer] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todosToShow, setTodosToShow] = useState([...todosFromTheServer]);
+  const [currentFilter, setCurrentFilter] = useState('ALL');
+  const [isAdding, setIsAdding] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [deletingCompleted, setDelitingCompleted] = useState(false);
+  const [customTodo, setCustomTodo] = useState<CustomTodo | null>(null);
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const filterTodos = (type: string) => {
+    setCurrentFilter(type);
+    switch (type) {
+      case 'COMPLETED': {
+        const completedTodos = todosFromTheServer.filter(
+          todo => todo.completed,
+        );
+
+        setTodosToShow(completedTodos);
+        break;
+      }
+
+      case 'ACTIVE': {
+        const activeTodos = todosFromTheServer.filter(
+          todo => !todo.completed,
+        );
+
+        setTodosToShow(activeTodos);
+        break;
+      }
+
+      default: {
+        setTodosToShow(todosFromTheServer);
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    getTodos(user.id)
+      .then(gotTodos => {
+        setTodosFromTheServer(gotTodos);
+        setTodosToShow([...gotTodos]);
+      })
+      .catch(() => setErrorMessage('Unable to get todos'));
+  }, [user]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    filterTodos(currentFilter);
+  }, [todosFromTheServer]);
+
+  const onFormSubmit = (value: string) => {
+    setIsAdding(true);
+
+    if (!value.trim) {
+      setErrorMessage("Title can't be empty");
+
+      return;
     }
-  }, []);
+
+    if (!user) {
+      return;
+    }
```

Comment:
I think you'll have a user in any case 🤔 You can't show the form if a user is not login